services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowHistorySpec.groovy

```diff
@@ -0,0 +1,140 @@
+package org.openkilda.functionaltests.spec.flows
+
+import static org.openkilda.testing.Constants.NON_EXISTENT_FLOW_ID
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""Verify that history records are created for the create/update actions.
+History record is created in case the create/update action is completed successfully.""")
+class FlowHistorySpec extends BaseSpecification {
+    String createAction = "Flow creating"
+    String createHistoryAction = "Created the flow"
+    String updateAction = "Flow updating"
+    String updateHistoryAction = "Updated the flow"
+
+    @Shared
+    Long timestampBefore
+
+    def setupOnce() {
+        timestampBefore = (System.currentTimeMillis() / 1000L) - 1000
```

Comment:
You can use `System.currentTimeSeconds()` instead of `(System.currentTimeMillis() / 1000L) `. Also, why do you subtract 1000?

Reply:
just to mitigate error in timestamp (on a local machine and on a system server)

Reply:
1000 seconds = 16.6 minutes. Is there really a 16min error possibility?

Reply:
I thought it is millisecond.
Fixed.